game/src/contrib/systems/HealthSystem.java

```diff
@@ -19,16 +21,22 @@
  */
 public class HealthSystem extends System {
 
-    // private record to hold all data during streaming
-    private record HSData(Entity e, HealthComponent hc, DrawComponent ac) {}
+    public HealthSystem() {
+        super(HealthComponent.class, getSet());
+    }
+
+    private static Set<Class<? extends Component>> getSet() {
+        Set<Class<? extends Component>> set = new HashSet<>();
+        set.add(DrawComponent.class);
+        return set;
+    }
```

Comment:
Brauchen wir da wirklich ne extra Methode dafür? Im Prinzip würde doch sowas in der Art ausreichen: 

```java
// Anfang der Klasse
private final Set<Class<? extends Component>> additionalComponents = Set.of(DrawComponent.class);

// Im Ctor
super(HealthComponent, additionalComponents);
```

Reply:
Eigentlich könnte man das auch direkt in die (geerbten) Attribute `keyComponent` und `additionalComponents` setzen, d.h. einmal den Super-Ctor und danach zwei Zuweisungen an die Attribute. Dazu müssten die aber `protected` und nicht `final` sein in `System`.

Reply:
Edit: Ich hatte einen Out-of-Coffee-Error. Folgendes geht:

```java
    public HealthSystem() {
        super(HealthComponent.class, Set.of(DrawComponent.class);
    }
```

D.h. man braucht keine statische Hilfsmethode und auch keine statische Hilfsvariable. Aus irgendeinem Grund hat das vorhin nicht geklappt, und das hat mir keine Ruhe gelassen. Tja, und jetzt ist auch IJ der Meinung, dass das ginge :)

Reply:
Edit 2: Man könnte statt dem `Set` auch mit varargs arbeiten: 

```java
// System
    public System(
            Class<? extends Component> keyComponent,
            Class<? extends Component>... additionalComponents) {
        ...
        this.additionalComponents = Set.of(additionalComponents);
        ...
    }

// ProjectileSystem
    public ProjectileSystem() {
        super(ProjectileComponent.class, PositionComponent.class, VelocityComponent.class);
    }
```

Reply:
Edit 2 ist super....

Reply:
und umgesetzt